lib_eio/path.ml

```diff
@@ -30,10 +31,22 @@ let with_lines path fn =
   let buf = Buf_read.of_flow flow ~max_size:max_int in
   fn (Buf_read.lines buf)
 
-let load path =
-  with_open_in path @@ fun flow ->
-  let buf = Buf_read.of_flow flow ~max_size:max_int in
-  Buf_read.take_all buf
+let load ((t:#Fs.dir), path) =
+  let st = t#stat path in
+  let mymax = min Sys.max_string_length Sys.max_array_length in
```

Comment:
Because we have to store it in a Cstruct.t before the converting it to a string, and max_array_length < max_string_length, at least on amd64.

Reply:
As far as I know, that doesn't apply to bigarrays like cstruct.

Reply:
Also, it occurs to me that, since we're opening the file anyway, it would be simpler to use `File.stat` here, not `Path.stat`.

Reply:
I did check and bigarray is constrained by the that limit, but I might have been mistaken I'll check again.

I move it to `File`, problem is I can't call it one unix domain socket, I've added the method to `File.ro`, to have it in socket it would have to be deeper, I'm not sure what is preferred here ?

Reply:
> I did check and bigarray is constrained by the that limit, but I might have been mistaken I'll check again.

Could you post a link to the code. It seems very surprising to me that a "big array" is the same size as a normal one.

In particular, https://v2.ocaml.org/api/Bigarray.html says:

> Bigarrays are not limited in size, unlike OCaml arrays.

Reply:
> > I did check and bigarray is constrained by the that limit, but I might have been mistaken I'll check again.
> 
> Could you post a link to the code. It seems very surprising to me that a "big array" is the same size as a normal one.
> 
> In particular, https://v2.ocaml.org/api/Bigarray.html says:
> 
> > Bigarrays are not limited in size, unlike OCaml arrays.

You're absolutely correct, I think my confusion came from the fact that Bigarray was using Array underneath and I assumed the same, but it's not:

`caml_make_vect` (Array.create) checks for `Max_wosize` where Bigarray's `caml_ba_create` does not. I've removed the test.